pkg/controller/worker/reconciler_reconcile.go

```diff
@@ -0,0 +1,334 @@
+	}
+
+	// Get the list of all existing machine deployments.
+	existingMachineDeployments := &machinev1alpha1.MachineDeploymentList{}
+	if err := r.client.List(ctx, &client.ListOptions{Namespace: worker.Namespace}, existingMachineDeployments); err != nil {
+		return err
+	}
+
+	// Generate machine deployment configuration based on previously computed list of deployments and deploy them.
+	r.logger.Info("Deploying the machine deployments", "worker", fmt.Sprintf("%s/%s", worker.Namespace, worker.Name))
+	if err := r.deployMachineDeployments(ctx, cluster, worker, existingMachineDeployments, wantedMachineDeployments, workerContext.MachineClassKind(), clusterAutoscalerRequired); err != nil {
+		return fmt.Errorf("Failed to generate the machine deployment config: '%s'", err.Error())
+	}
+
+	// Wait until all generated machine deployments are healthy/available.
+	if err := r.waitUntilMachineDeploymentsAvailable(ctx, cluster, worker, wantedMachineDeployments); err != nil {
+		return gardencorev1alpha1helper.DetermineError(fmt.Sprintf("Failed while waiting for all machine deployments to be ready: '%s'", err.Error()))
+	}
+
+	// Delete all old machine deployments (i.e. those which were not previously computed but exist in the cluster).
+	if err := r.cleanupMachineDeployments(ctx, existingMachineDeployments, wantedMachineDeployments); err != nil {
+		return fmt.Errorf("Failed to cleanup the machine deployments: '%s'", err.Error())
+	}
+
+	// Delete all old machine classes (i.e. those which were not previously computed but exist in the cluster).
+	if err := r.cleanupMachineClasses(ctx, worker.Namespace, workerContext.MachineClassList(), wantedMachineDeployments); err != nil {
+		return fmt.Errorf("Failed to cleanup the machine classes: '%s'", err.Error())
+	}
+
+	// Delete all old machine class secrets (i.e. those which were not previously computed but exist in the cluster).
+	if err := r.cleanupMachineClassSecrets(ctx, worker.Namespace, wantedMachineDeployments); err != nil {
+		return fmt.Errorf("Failed to cleanup the orphaned machine class secrets: '%s'", err.Error())
+	}
+
+	// Scale down machine-controller-manager if shoot is hibernated.
+	if controller.IsHibernated(cluster.Shoot) {
+		if err := util.ScaleDeployment(ctx, r.client, worker.Namespace, machineControllerManagerName, 0); err != nil {
+			return err
+		}
+	}
+
+	if err := r.updateWorkerStatus(ctx, worker, wantedMachineDeployments); err != nil {
+		return fmt.Errorf("Failed to update the status in the Worker resource: %+v", err)
+	}
+
+	return nil
+}
+
+func (r *reconciler) deployMachineDeployments(ctx context.Context, cluster *controller.Cluster, worker *extensionsv1alpha1.Worker, existingMachineDeployments *machinev1alpha1.MachineDeploymentList, wantedMachineDeployments MachineDeployments, classKind string, clusterAutoscalerRequired bool) error {
+	for _, deployment := range wantedMachineDeployments {
+		var (
+			labels                    = map[string]string{"name": deployment.Name}
```

Comment:
What do you mean? Let's not change this now, we have hundreds of clusters with this label and I don't know which side-effects prefixing this might cause now (if it's even possible).